server/admin_management/api/app/routers/router_guards.py

```diff
@@ -121,71 +115,140 @@ async def get_target_user_from_id(
+        raaid = request.path_params["application_admin_id"]
+        LOGGER.debug("Dependency 'get_target_user_from_id': 'application_admin_id' " +
+                     f"path param found: {raaid}.")
         application_admin = application_admin_service.get_application_admin_by_id(
-            request.path_params["application_admin_id"]
-        )
-        return (
-            TargetUser.model_validate(application_admin.user)
-            if application_admin is not None
-            else None
+            raaid
         )
+        if application_admin is not None:
+            return TargetUser.model_validate(application_admin.user)
+        else:
+            error_msg = "Parameter 'application_admin_id' {raaid} is missing or invalid."
+            utils.raise_http_exception(
+                error_code=ERROR_CODE_INVALID_REQUEST_PARAMETER,
+                error_msg=error_msg
+            )
+
     elif "access_control_privilege_id" in request.path_params:
+        acpid = request.path_params["access_control_privilege_id"]
+        LOGGER.debug("Dependency 'get_target_user_from_id': 'access_control_privilege_id' " +
+                     f"path param found: {acpid}.")
         access_control_privilege = access_control_privilege_service.get_acp_by_id(
-            request.path_params["access_control_privilege_id"]
-        )
-        return (
-            TargetUser.model_validate(access_control_privilege.user)
-            if access_control_privilege is not None
-            else None
+            acpid
         )
+        if access_control_privilege is not None:
+            return TargetUser.model_validate(access_control_privilege.user)
+        else:
+            error_msg = "Parameter 'access_control_privilege_id' {acpid} is missing or invalid."
+            utils.raise_http_exception(
+                error_code=ERROR_CODE_INVALID_REQUEST_PARAMETER,
+                error_msg=error_msg
+            )
     else:
         # from body - {user_name/user_type_code}, when grant admin access
-        try:
-            rbody = await request.json()
-            user = user_service.get_user_by_domain_and_name(
-                rbody["user_type_code"],
-                rbody["user_name"],
+        rbody = await request.json()
+        LOGGER.debug(f"Dependency 'get_target_user_from_id' request body {rbody}.")
+        rb_user_name = rbody["user_name"]
+        rb_user_type_code = rbody["user_type_code"]
+        # user_guid is searched from IDIM search and passed from frontend.
+        rb_user_guid = rbody["user_guid"]
+
+        user = user_service.get_user_by_domain_and_name(
+            rb_user_type_code,
+            rb_user_name,
+        )
+        if user is not None:
+            found_target_user = TargetUser.model_validate(user)
+            # old data might not have user_guid.
+            if found_target_user.user_guid is None:
+                found_target_user.user_guid = rb_user_guid
```

Comment:
Hi Ian, I think later we'll change to get user by user_guid when search the target user in our database, but for now it's okay, just think in this case, no matter `found_target_user.user_guid` is None or not none, we should always set `found_target_user.user_guid = rb_user_guid`. Because the user_guid from the parameter will be the one we should rely on.

Reply:
Hm...are we going to rely on "**user_guid**" passed from frontend? Maybe I missed the discussion. If the user from the database already has a user_guid (which is more reliable), we shouldn't update it again anymore.
I am not sure if relying on "user_guid" would introduce a hole in system, maybe I was too paranoid?
I am thinking we can change this logic later when we do have validation and getting user from user_guid in place.